#### Introduction: 
Coronavirus disease (COVID-19) is an  PANDEMIC infectious disease caused by severe acute respiratory syndrome coronavirus 2 (SARS‑CoV‑2). This new virus and disease were unknown before the outbreak began in Wuhan, China, in December 2019. COVID-19 is a pandemic affecting many countries globally.The time between exposure to COVID-19 and the moment when symptoms start is commonly around five to six days but can range from 1 – 14 days. I have gathered Covid-19 data from different sources. Dataset contains confirmed case, deaths and recovered cases, new active cases in world and in India and indian  State/UnionTerritory separately.

 * Coronavirus Analysis based on Confirmed, Dead and Recovered Patients all over World.
 * Second Analysis based on Deaths & Recovered Patient in India and its States and Union territories.
 * Analysis based on State Testing. 
 * Analysis based on Time series.


In [354]:
import pandas as pd
import numpy  as np
import matplotlib.pyplot as plt
import seaborn as sns
from timeit import default_timer as timer
#from lxml import html 
#import re
import requests
import json

In [355]:
import matplotlib.ticker as ticker
import statsmodels.api as sm
import statsmodels.formula.api as smf
import plotly.express as px
import datetime
from plotly.subplots import make_subplots
import requests
from bs4 import BeautifulSoup 
import plotly.io as pio
from pandas.io.json import json_normalize
pio.renderers.default='notebook'
import plotly.graph_objects as go

### Covid-19 data gathered from different sources:
    1. On the basis of all countries: data scraped using beautiful soup https://www.worldometers.info/coronavirus/
    2. Covid analysis of India:'covid_19_india.csv'    
    3. Analysis based on Age:'AgeGroupDetails.csv'
    4. Analysis based on ICMR Testing labs:'ICMRTestingDetails.csv'from    
    5. Analysis based on testing done in indian states:'StatewiseTestingDetails.csv' https://www.kaggle.com/sudalairajkumar/covid19-in-india
    6. Analysis based on Time Series Confirmed cases all over world from :
    7. Analysis based on Time Series Recovered case all over world:'time_series_covid19_recovered.csv'.
    8. Analysis based on Time Series Deaths case all over world:'time_series_covid19_deaths.csv'
    https://data.humdata.org/dataset/5dff64bc-a671-48da-aa87-2ca40d7abf02

### Covid_19 All countries scraped using beautiful soup from worldometers.com: 
    
    * The coronavirus COVID-19 is affecting 212 countries and territories around the world and 2 international conveyances data.
    
        - Total_cases
        - Total_deaths
        - Total_Recovered
        - New_cases
        - New Deaths
        - Active_Cases
        - Serious_Case
        - TotCases/1Mpop
        - Deaths/1Mpop
        - Total Tests
        - Test/1Mpop

In [356]:
url = "https://www.worldometers.info/coronavirus/" 
req_data = requests.get(url)
soup = BeautifulSoup(req_data.text, 'html.parser') 

# x = soup.findAll("tbody")
# if x is not None and len(x) > 0:
#     section = x[0]

table = soup.find('table', attrs={'id': 'main_table_countries_today'})
header = [col_name.text.rstrip('\n').strip() for col_name in table.select('thead th')]
table_rows = table.find_all('tr')
data = []
for tr in table_rows:
    try: 
        td = tr.find_all('td')
        row = [tr.text for tr in td]
        print(row) 
        data.append(row)
        full_data = pd.read_html(str(table))[0]
    except requests.Response.raise_for_status() as e:
        print("Error: Invalid Response Error.")

[]
['\nNorth America\n', '1,386,646', '+10,745', '83,462', '+1,035', '265,196', '1,037,988', '16,999', '', '', '', '', 'North America']
['\nEurope\n', '1,552,373', '+27,469', '148,189', '+1,558', '609,359', '794,825', '15,652', '', '', '', '', 'Europe']
['\nAsia\n', '614,695', '+11,630', '21,079', '+273', '336,505', '257,111', '5,201', '', '', '', '', 'Asia']
['\nSouth America\n', '254,006', '+2,395', '12,851', '+26', '88,906', '152,249', '9,981', '', '', '', '', 'South America']
['\nAfrica\n', '53,821', '+1,108', '2,047', '+32', '18,770', '33,004', '170', '', '', '', '', 'Africa']
['\nOceania\n', '8,489', '+22', '118', '', '7,467', '904', '26', '', '', '', '', 'Australia/Oceania']
['\n\n', '721', '', '15', '', '645', '61', '4', '', '', '', '', '']
['World', '3,870,751', '+53,369', '267,761', '+2,924', '1,326,848', '2,276,142', '48,033', '497', '34.4', '', '', 'All']
['USA', '1,271,007', '+7,915', '75,558 ', '+759', '213,562', '981,887', '15,827', '3,840', '228', '8,081,734', '24,416',

['Greece', '2,678', '+15', '148 ', '+1', '1,374', '1,156', '33', '257', '14', '90,043', '8,639', 'Europe']
['Iraq', '2,480', '', '102 ', '', '1,602', '776', '', '62', '3', '117,726', '2,927', 'Asia']
['Uzbekistan', '2,269', '+36', '10 ', '', '1,656', '603', '8', '68', '0.3', '325,000', '9,710', 'Asia']
['Cameroon', '2,267', '+2', '108 ', '', '1,002', '1,157', '12', '85', '4', '', '', 'Africa']
['Azerbaijan', '2,204', '+77', '28 ', '', '1,551', '625', '18', '217', '3', '175,910', '17,350', 'Asia']
['Croatia', '2,125', '+6', '86 ', '+1', '1,641', '398', '14', '518', '21', '41,938', '10,216', 'Europe']
['Bosnia and Herzegovina', '2,027', '+40', '90 ', '+4', '954', '983', '4', '618', '27', '38,187', '11,639', 'Europe']
['Bolivia', '1,886', '+84', '91 ', '+5', '198', '1,597', '3', '162', '8', '7,651', '655', 'South America']
['Guinea', '1,856', '', '11 ', '', '597', '1,248', '', '141', '0.8', '', '', 'Africa']
['Bulgaria', '1,829', '+51', '84 ', '', '384', '1,361', '43', '263', '12', '52,93

['Bermuda', '118', '', '7 ', '', '59', '52', '4', '1,895', '112', '3,569', '57,308', 'North America']
['Trinidad and Tobago', '116', '', '8 ', '', '103', '5', '', '83', '6', '2,154', '1,539', 'North America']
['Haiti', '108', '+7', '12 ', '', '15', '81', '', '9', '1', '1,091', '96', 'North America']
['Aruba', '101', '', '2 ', '', '89', '10', '4', '946', '19', '1,798', '16,841', 'North America']
['Nepal', '101', '+2', ' ', '', '22', '79', '', '3', '', '67,066', '2,302', 'Asia']
['Uganda', '100', '', ' ', '', '55', '45', '', '2', '', '47,620', '1,041', 'Africa']
['Monaco', '95', '', '4 ', '', '82', '9', '1', '2,421', '102', '', '', 'Europe']
['CAR', '94', '', ' ', '', '10', '84', '', '19', '', '3,498', '724', 'Africa']
['Guyana', '93', '', '10 ', '', '27', '56', '3', '118', '13', '714', '908', 'South America']
['Bahamas', '92', '', '11 ', '', '26', '55', '1', '234', '28', '1,500', '3,814', 'North America']
['Barbados', '82', '', '7 ', '', '47', '28', '4', '285', '24', '2,629', '9,148', '

In [357]:
full_data.head(10)

Country,Other  TotalCases NewCases  TotalDeaths NewDeaths  TotalRecovered  \
0         World     3870751  +53,369     267761.0    +2,924       1326848.0   
1           USA     1271007   +7,915      75558.0      +759        213562.0   
2         Spain      256855   +3,173      26070.0      +213        163919.0   
3         Italy      215858   +1,401      29958.0      +274         96276.0   
4            UK      206715   +5,614      30615.0      +539             NaN   
5        Russia      177160  +11,231       1625.0       +88         23803.0   
6        France      174191      NaN      25809.0       NaN         53972.0   
7       Germany      168912     +750       7336.0       +61        139900.0   
8        Turkey      133721   +1,977       3641.0       +57         82984.0   
9        Brazil      127389     +778       8605.0       +17         51370.0   

   ActiveCases  Serious,Critical  Tot Cases/1M pop  Deaths/1M pop  TotalTests  \
0      2276142           48033.0             497.0           34.4         NaN   
1       981887           15827.0            3840.0          228.0   8081734.0   
2        66866            2075.0            5494.0          558.0   1932455.0   
3        89624            1311.0            3570.0          495.0   2381288.0   
4       175756            1559.0            3045.0          451.0   1448010.0   
5       151732            2300.0            1214.0           11.0   4803192.0   
6        94410            3147.0            2669.0          395.0   1100228.0   
7        21676            1884.0            2016.0           88.0   2755770.0   
8        47096            1260.0            1586.0           43.0   1265119.0   
9        67414            8318.0             599.0           40.0    339552.0   

   Tests/ 1M pop  
0            NaN  
1        24416.0  
2        41332.0  
3        39385.0  
4        21330.0  
5        32913.0  
6        16856.0  
7        32891.0  
8        15000.0  
9         1597.0

### Covid_19 India CSV file :
- Confirmed cases, Deaths, Cured cases.
-Date:  30/01/2 till 06/05/20	


In [358]:
covid_india = pd.read_csv('covid_19_india.csv')

In [359]:
covid_india.head(5)

Sno      Date     Time State/UnionTerritory ConfirmedIndianNational  \
0    1  30/01/20  6:00 PM               Kerala                       1   
1    2  31/01/20  6:00 PM               Kerala                       1   
2    3  01/02/20  6:00 PM               Kerala                       2   
3    4  02/02/20  6:00 PM               Kerala                       3   
4    5  03/02/20  6:00 PM               Kerala                       3   

  ConfirmedForeignNational  Cured  Deaths  Confirmed  
0                        0      0       0          1  
1                        0      0       0          1  
2                        0      0       0          2  
3                        0      0       0          3  
4                        0      0       0          3

*** 30/jan/2020 india reported first case in kerala 
*** 02/march/2020 india reported their covid_19 cases i Telengana ,Delhi and after that  covid_19 cases reported to in other state and /UnionTerritory.
*** 13/03/2020 Karnataka reported death of one patients due to covid_19

In [360]:
covid_india_age = pd.read_csv('AgeGroupDetails.csv')
covid_india_age

Sno AgeGroup  TotalCases Percentage
0    1      0-9          22      3.18%
1    2    10-19          27      3.90%
2    3    20-29         172     24.86%
3    4    30-39         146     21.10%
4    5    40-49         112     16.18%
5    6    50-59          77     11.13%
6    7    60-69          89     12.86%
7    8    70-79          28      4.05%
8    9     >=80          10      1.45%
9   10  Missing           9      1.30%

In [361]:
covid_india_testing = pd.read_csv('ICMRTestingDetails.csv')
covid_india_testing.tail(5)

SNo       DateTime  TotalSamplesTested  TotalIndividualsTested  \
37   38  23/04/20 9:00            541789.0                525667.0   
38   39  24/04/20 9:00            579957.0                     NaN   
39   40  25/04/20 9:00            625309.0                     NaN   
40   41  26/04/20 9:00            665819.0                     NaN   
41   42  27/04/20 9:00            716733.0                     NaN   

    TotalPositiveCases  
37             23502.0  
38                 NaN  
39                 NaN  
40                 NaN  
41                 NaN

### Covid_19 India testing CSV file :
- CSV contain Statewise Testing Positive and Negative Patients data. 

In [362]:
covid_india_state_testing = pd.read_csv('StatewiseTestingDetails.csv')
covid_india_state_testing

Date                        State  TotalSamples  Negative  Positive
0    2020-04-17  Andaman and Nicobar Islands        1403.0    1210.0      12.0
1    2020-04-24  Andaman and Nicobar Islands        2679.0       NaN      27.0
2    2020-04-27  Andaman and Nicobar Islands        2848.0       NaN      33.0
3    2020-05-01  Andaman and Nicobar Islands        3754.0       NaN      33.0
4    2020-04-02               Andhra Pradesh        1800.0    1175.0     132.0
..          ...                          ...           ...       ...       ...
754  2020-04-30                  West Bengal       16525.0       NaN     758.0
755  2020-05-01                  West Bengal       18566.0       NaN       NaN
756  2020-05-02                  West Bengal       20976.0       NaN     795.0
757  2020-05-03                  West Bengal       22915.0       NaN     922.0
758  2020-05-04                  West Bengal       25116.0       NaN    1259.0

[759 rows x 5 columns]

#### Time Series Covid_19 case Csv file :
-TimeSeries data contain DATE and Country information.

In [363]:
covid_time_series_C= pd.read_csv('time_series_covid19_confirmed.csv')
covid_time_series_C

Province/State         Country/Region        Lat       Long  \
0                          NaN            Afghanistan  33.000000  65.000000   
1                          NaN                Albania  41.153300  20.168300   
2                          NaN                Algeria  28.033900   1.659600   
3                          NaN                Andorra  42.506300   1.521800   
4                          NaN                 Angola -11.202700  17.873900   
..                         ...                    ...        ...        ...   
259  Saint Pierre and Miquelon                 France  46.885200 -56.315900   
260                        NaN            South Sudan   6.877000  31.307000   
261                        NaN         Western Sahara  24.215500 -12.885800   
262                        NaN  Sao Tome and Principe   0.186360   6.613081   
263                        NaN                  Yemen  15.552727  48.516388   

     1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  4/20/20  \
0          0        0        0        0        0        0  ...     1026   
1          0        0        0        0        0        0  ...      584   
2          0        0        0        0        0        0  ...     2718   
3          0        0        0        0        0        0  ...      717   
4          0        0        0        0        0        0  ...       24   
..       ...      ...      ...      ...      ...      ...  ...      ...   
259        0        0        0        0        0        0  ...        1   
260        0        0        0        0        0        0  ...        4   
261        0        0        0        0        0        0  ...        6   
262        0        0        0        0        0        0  ...        4   
263        0        0        0        0        0        0  ...        1   

     4/21/20  4/22/20  4/23/20  4/24/20  4/25/20  4/26/20  4/27/20  4/28/20  \
0       1092     1176     1279     1351     1463     1531     1703     1828   
1        609      634      663      678      712      726      736      750   
2       2811     2910     3007     3127     3256     3382     3517     3649   
3        717      723      723      731      738      738      743      743   
4         24       25       25       25       25       26       27       27   
..       ...      ...      ...      ...      ...      ...      ...      ...   
259        1        1        1        1        1        1        1        1   
260        4        4        5        5        5        6        6       34   
261        6        6        6        6        6        6        6        6   
262        4        4        4        4        4        4        4        8   
263        1        1        1        1        1        1        1        1   

     4/29/20  
0       1939  
1        766  
2       3848  
3        743  
4         27  
..       ...  
259        1  
260       34  
261        6  
262        8  
263        6  

[264 rows x 103 columns]

In [364]:
covid_time_series_covid_19_R = pd.read_csv('time_series_covid19_recovered.csv')
covid_time_series_covid_19_R

Province/State         Country/Region        Lat       Long  \
0                          NaN            Afghanistan  33.000000  65.000000   
1                          NaN                Albania  41.153300  20.168300   
2                          NaN                Algeria  28.033900   1.659600   
3                          NaN                Andorra  42.506300   1.521800   
4                          NaN                 Angola -11.202700  17.873900   
..                         ...                    ...        ...        ...   
245  Saint Pierre and Miquelon                 France  46.885200 -56.315900   
246                        NaN            South Sudan   6.877000  31.307000   
247                        NaN         Western Sahara  24.215500 -12.885800   
248                        NaN  Sao Tome and Principe   0.186360   6.613081   
249                        NaN                  Yemen  15.552727  48.516388   

     1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  4/20/20  \
0          0        0        0        0        0        0  ...      135   
1          0        0        0        0        0        0  ...      327   
2          0        0        0        0        0        0  ...     1099   
3          0        0        0        0        0        0  ...      248   
4          0        0        0        0        0        0  ...        6   
..       ...      ...      ...      ...      ...      ...  ...      ...   
245        0        0        0        0        0        0  ...        0   
246        0        0        0        0        0        0  ...        0   
247        0        0        0        0        0        0  ...        0   
248        0        0        0        0        0        0  ...        0   
249        0        0        0        0        0        0  ...        0   

     4/21/20  4/22/20  4/23/20  4/24/20  4/25/20  4/26/20  4/27/20  4/28/20  \
0        150      166      179      188      188      207      220      228   
1        345      356      385      394      403      410      422      431   
2       1152     1204     1355     1408     1479     1508     1558     1651   
3        282      309      333      344      344      344      385      398   
4          6        6        6        6        6        6        6        6   
..       ...      ...      ...      ...      ...      ...      ...      ...   
245        0        0        0        0        0        0        0        0   
246        0        0        0        0        0        0        0        0   
247        0        0        5        5        5        5        5        5   
248        0        0        0        0        0        0        0        4   
249        0        0        0        1        1        1        1        1   

     4/29/20  
0        252  
1        455  
2       1702  
3        423  
4          7  
..       ...  
245        0  
246        0  
247        5  
248        4  
249        1  

[250 rows x 103 columns]

In [365]:
covid_time_series_D = pd.read_csv('time_series_covid19_deaths.csv')
covid_time_series_D

Province/State         Country/Region        Lat       Long  \
0                          NaN            Afghanistan  33.000000  65.000000   
1                          NaN                Albania  41.153300  20.168300   
2                          NaN                Algeria  28.033900   1.659600   
3                          NaN                Andorra  42.506300   1.521800   
4                          NaN                 Angola -11.202700  17.873900   
..                         ...                    ...        ...        ...   
259  Saint Pierre and Miquelon                 France  46.885200 -56.315900   
260                        NaN            South Sudan   6.877000  31.307000   
261                        NaN         Western Sahara  24.215500 -12.885800   
262                        NaN  Sao Tome and Principe   0.186360   6.613081   
263                        NaN                  Yemen  15.552727  48.516388   

     1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  4/20/20  \
0          0        0        0        0        0        0  ...       36   
1          0        0        0        0        0        0  ...       26   
2          0        0        0        0        0        0  ...      384   
3          0        0        0        0        0        0  ...       37   
4          0        0        0        0        0        0  ...        2   
..       ...      ...      ...      ...      ...      ...  ...      ...   
259        0        0        0        0        0        0  ...        0   
260        0        0        0        0        0        0  ...        0   
261        0        0        0        0        0        0  ...        0   
262        0        0        0        0        0        0  ...        0   
263        0        0        0        0        0        0  ...        0   

     4/21/20  4/22/20  4/23/20  4/24/20  4/25/20  4/26/20  4/27/20  4/28/20  \
0         36       40       42       43       47       50       57       58   
1         26       27       27       27       27       28       28       30   
2        392      402      407      415      419      425      432      437   
3         37       37       37       40       40       40       40       41   
4          2        2        2        2        2        2        2        2   
..       ...      ...      ...      ...      ...      ...      ...      ...   
259        0        0        0        0        0        0        0        0   
260        0        0        0        0        0        0        0        0   
261        0        0        0        0        0        0        0        0   
262        0        0        0        0        0        0        0        0   
263        0        0        0        0        0        0        0        0   

     4/29/20  
0         60  
1         30  
2        444  
3         42  
4          2  
..       ...  
259        0  
260        0  
261        0  
262        0  
263        0  

[264 rows x 103 columns]

## Assessing Data:

****Assessing Full_data of covid_19 all countries Data 

In [366]:
full_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 216 entries, 0 to 215
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Country,Other     216 non-null    object 
 1   TotalCases        216 non-null    int64  
 2   NewCases          105 non-null    object 
 3   TotalDeaths       179 non-null    float64
 4   NewDeaths         66 non-null     object 
 5   TotalRecovered    208 non-null    float64
 6   ActiveCases       216 non-null    int64  
 7   Serious,Critical  135 non-null    float64
 8   Tot Cases/1M pop  214 non-null    float64
 9   Deaths/1M pop     177 non-null    float64
 10  TotalTests        182 non-null    float64
 11  Tests/ 1M pop     182 non-null    float64
dtypes: float64(7), int64(2), object(3)
memory usage: 20.4+ KB


The coronavirus COVID-19 has affected 216 countries and territories around the world and 2 international conveyances are reported in www.worldometers.info/coronavirus/.
 1.Total 216 countries, 177 countries reported deaths.
 2 TotalRecovered cases reported in 207 countries.
 3.61 NewDeaths are reported.
 4.Testing reported in 177 countries.
** Some data is missing but can't analyze missing data may be  True or confirmed cases,deaths,Recovered cases are not reported .

#### Data Issue 1:
***Rename Columns:

In [367]:
full_data.rename(columns = lambda X:X.strip().lower().replace(" ","_"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("country,other","country"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("totalcases","total_cases"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("newcases","new_cases"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("totaldeaths","total_deaths"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("newdeaths","new_deaths"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("totalrecovered","total_recovered"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("activecases","active_cases"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("serious,critical","serious"),inplace =True)
full_data.rename(columns = lambda X:X.strip().lower().replace("totaltests","total_tests"),inplace =True)


In [368]:
full_data.rename(columns = lambda X:X.strip().lower().replace("tot cases/1m_pop","totcases/1m_pop"),inplace =True)


#### Data Issue 2:
##### Fill NaN valeus in numeric fields with zero:

In [369]:
full_data.fillna(0, inplace=True)

#### Data  issues:3
    Replace comma and plus sign from all the value in dataframe column. 
    
#### Data Issue 4:
    datatype Object is converted into integer value datatype

In [370]:
full_data['total_deaths'] = full_data['total_deaths'].astype('int64')
full_data['total_recovered'] = full_data['total_recovered'].astype('int64')
full_data['serious'] = full_data['serious'].astype('int64')
full_data['deaths/1m_pop'] = full_data['deaths/1m_pop'].astype('int64')
full_data['total_tests'] = full_data['total_tests'].astype('int64')
full_data['tests/_1m_pop'] = full_data['tests/_1m_pop'].astype('int64')


In [371]:
full_data['new_cases'] = full_data['new_cases'].str.replace(',', '', regex=True)
full_data['new_deaths'] = full_data['new_deaths'].str.replace(',','', regex=True)
full_data['new_cases'].fillna(0,inplace=True)
full_data['new_deaths'].fillna(0,inplace=True)
full_data['new_cases'] = full_data['new_cases'].astype('int64')
full_data['new_deaths'] = full_data['new_deaths'].astype('int64')
full_data

country  total_cases  new_cases  total_deaths  new_deaths  \
0                    World      3870751      53369        267761        2924   
1                      USA      1271007       7915         75558         759   
2                    Spain       256855       3173         26070         213   
3                    Italy       215858       1401         29958         274   
4                       UK       206715       5614         30615         539   
..                     ...          ...        ...           ...         ...   
211         Western Sahara            6          0             0           0   
212               Anguilla            3          0             0           0   
213  Saint Pierre Miquelon            1          0             0           0   
214                  China        82885          2          4633           0   
215                 Total:      3870751      53369        267761        2924   

     total_recovered  active_cases  serious  tot cases/1m_pop  deaths/1m_pop  \
0            1326848       2276142    48033             497.0             34   
1             213562        981887    15827            3840.0            228   
2             163919         66866     2075            5494.0            558   
3              96276         89624     1311            3570.0            495   
4                  0        175756     1559            3045.0            451   
..               ...           ...      ...               ...            ...   
211                5             1        0              10.0              0   
212                3             0        0             200.0              0   
213                0             1        0             173.0              0   
214            77957           295       23              58.0              3   
215          1326848       2276142    48033             496.6             34   

     total_tests  tests/_1m_pop  
0              0              0  
1        8081734          24416  
2        1932455          41332  
3        2381288          39385  
4        1448010          21330  
..           ...            ...  
211            0              0  
212            0              0  
213            0              0  
214            0              0  
215            0              0  

[216 rows x 12 columns]

In [372]:
#full_data['new_cases'] = full_data['new_cases'].Str.replace('+','', regex=True)
#full_data['new_deaths'] = full_data['new_deaths'].str.replace('+','', regex=True)

In [373]:
full_data.describe()

total_cases     new_cases   total_deaths   new_deaths  \
count  2.160000e+02    216.000000     216.000000   216.000000   
mean   5.376043e+04    741.236111    3718.902778    40.611111   
std    3.814252e+05   5208.921604   26383.181598   287.893418   
min    1.000000e+00      0.000000       0.000000     0.000000   
25%    9.375000e+01      0.000000       2.000000     0.000000   
50%    7.035000e+02      0.000000      13.500000     0.000000   
75%    5.579250e+03     57.250000     123.250000     1.250000   
max    3.870751e+06  53369.000000  267761.000000  2924.000000   

       total_recovered  active_cases       serious  tot cases/1m_pop  \
count     2.160000e+02  2.160000e+02    216.000000        216.000000   
mean      1.842569e+04  3.161308e+04    667.125000        942.936574   
std       1.290179e+05  2.282714e+05   4762.536576       2052.394366   
min       0.000000e+00  0.000000e+00      0.000000          0.000000   
25%       2.550000e+01  2.750000e+01      0.000000         41.750000   
50%       2.460000e+02  2.960000e+02      3.000000        198.500000   
75%       1.747750e+03  2.409000e+03     24.000000        993.250000   
max       1.326848e+06  2.276142e+06  48033.000000      18331.000000   

       deaths/1m_pop   total_tests  tests/_1m_pop  
count     216.000000  2.160000e+02     216.000000  
mean       43.722222  1.969499e+05   14233.763889  
std       129.985926  7.175381e+05   25282.430674  
min         0.000000  0.000000e+00       0.000000  
25%         0.000000  7.215000e+02     389.500000  
50%         3.000000  1.170250e+04    3335.500000  
75%        24.250000  1.190432e+05   16957.500000  
max      1208.000000  8.081734e+06  170211.000000

In [374]:
full_data.sum()

country             WorldUSASpainItalyUKRussiaFranceGermanyTurkeyB...
total_cases                                                  11612253
new_cases                                                      160107
total_deaths                                                   803283
new_deaths                                                       8772
total_recovered                                               3979950
active_cases                                                  6828426
serious                                                        144099
tot cases/1m_pop                                               203674
deaths/1m_pop                                                    9444
total_tests                                                  42541175
tests/_1m_pop                                                 3074493
dtype: object

In [375]:
full_data.total_cases.max()

3870751

In [376]:
full_data.total_deaths.max()

267761

In [377]:
full_data.total_recovered.max()

1326848

In [378]:
full_data.duplicated().sum()

0

In [379]:
full_data.isna().sum()

country             0
total_cases         0
new_cases           0
total_deaths        0
new_deaths          0
total_recovered     0
active_cases        0
serious             0
tot cases/1m_pop    0
deaths/1m_pop       0
total_tests         0
tests/_1m_pop       0
dtype: int64

In [380]:
full_data['country'].value_counts()

Congo               1
Timor-Leste         1
UAE                 1
Greenland           1
China               1
                   ..
CAR                 1
Eritrea             1
Papua New Guinea    1
Canada              1
UK                  1
Name: country, Length: 216, dtype: int64

In [381]:
full_data.country.nunique()

216

### All Countries Latitude and Longitude  Data:
Some  latitude and longitude data is missing in countries_data.

In [382]:
countries = pd.read_csv('countries_data.csv', encoding= 'unicode_escape')
countries

country   latitude   longitude                   name
0                       AD  42.546245    1.601554                Andorra
1                       AE  23.424076   53.847818   United Arab Emirates
2                       AF  33.939110   67.709953            Afghanistan
3                       AG  17.060816  -61.796428    Antigua and Barbuda
4                       AI  18.220554  -63.068615               Anguilla
..                     ...        ...         ...                    ...
270             MS Zaandam  52.442039    4.829199             MS Zaandam
271  Caribbean Netherlands  12.178400     68.2385  Caribbean Netherlands
272              St. Barth  17.900000     62.8333              St. Barth
273  Saint Pierre Miquelon  46.885200     56.3159  Saint Pierre Miquelon
274                    CAR   6.611100     20.9394                    CAR

[275 rows x 4 columns]

In [383]:
def getLatL(country):
    row = countries.loc[countries['name'] == country ]
    return row.latitude.values


In [384]:
def getLong(country):
    row = countries.loc[countries['name'] == country]   
    return row.longitude.values

In [385]:
full_data['lat'] =full_data.apply(lambda row: getLatL(row['country']), axis=1)
full_data['lat'] = full_data['lat'].str.get(0)
full_data['long'] =full_data.apply(lambda row: getLong(row['country']), axis=1)
full_data['long']= full_data['long'].str.get(0)
full_data['long'] = full_data['long'].astype('float64') 
print(full_data.dtypes)
full_data.head()

country              object
total_cases           int64
new_cases             int64
total_deaths          int64
new_deaths            int64
total_recovered       int64
active_cases          int64
serious               int64
tot cases/1m_pop    float64
deaths/1m_pop         int64
total_tests           int64
tests/_1m_pop         int64
lat                 float64
long                float64
dtype: object


country  total_cases  new_cases  total_deaths  new_deaths  total_recovered  \
0   World      3870751      53369        267761        2924          1326848   
1     USA      1271007       7915         75558         759           213562   
2   Spain       256855       3173         26070         213           163919   
3   Italy       215858       1401         29958         274            96276   
4      UK       206715       5614         30615         539                0   

   active_cases  serious  tot cases/1m_pop  deaths/1m_pop  total_tests  \
0       2276142    48033             497.0             34            0   
1        981887    15827            3840.0            228      8081734   
2         66866     2075            5494.0            558      1932455   
3         89624     1311            3570.0            495      2381288   
4        175756     1559            3045.0            451      1448010   

   tests/_1m_pop        lat       long  
0              0        NaN        NaN  
1          24416  37.090240 -95.712891  
2          41332  40.463667  -3.749220  
3          39385  41.871940  12.567380  
4          21330  55.378100   3.436000

**** latitude and longitude data fix manually in excel. 
**** latitude, longitude data and full_data dataframe from worldometer combined in one table.

In [386]:
## Save the full_data  having  latitude and longitude data  in new csv file name covid.
full_data.to_csv('covid.csv')

In [387]:
# full_data['lat'] = full_data.apply(lambda row: getLatL(row['country']), axis=1)
# full_data['long'] = full_data.apply(lambda row: getLong(row['country']), axis=1)
# full_data.info()

**** Assessing Data of covid_19 India Data .

In [388]:
covid_india.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1671 entries, 0 to 1670
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Sno                       1671 non-null   int64 
 1   Date                      1671 non-null   object
 2   Time                      1671 non-null   object
 3   State/UnionTerritory      1671 non-null   object
 4   ConfirmedIndianNational   1671 non-null   object
 5   ConfirmedForeignNational  1671 non-null   object
 6   Cured                     1671 non-null   int64 
 7   Deaths                    1671 non-null   int64 
 8   Confirmed                 1671 non-null   int64 
dtypes: int64(4), object(5)
memory usage: 117.6+ KB


In [389]:
covid_india.describe()

Sno        Cured       Deaths     Confirmed
count  1671.000000  1671.000000  1671.000000   1671.000000
mean    836.000000    86.929982    13.054458    408.981448
std     482.520466   251.026352    49.430082   1188.132537
min       1.000000     0.000000     0.000000      0.000000
25%     418.500000     0.000000     0.000000      5.000000
50%     836.000000     5.000000     0.000000     32.000000
75%    1253.500000    34.000000     4.000000    258.500000
max    1671.000000  2819.000000   617.000000  15525.000000

In [390]:
covid_india['State/UnionTerritory'].value_counts()

Kerala                         98
Delhi                          66
Telengana                      66
Rajasthan                      65
Haryana                        64
Uttar Pradesh                  64
Ladakh                         61
Tamil Nadu                     61
Maharashtra                    59
Karnataka                      59
Jammu and Kashmir              59
Punjab                         59
Andhra Pradesh                 56
Uttarakhand                    53
Odisha                         52
Puducherry                     50
West Bengal                    50
Chhattisgarh                   49
Chandigarh                     49
Gujarat                        48
Himachal Pradesh               47
Madhya Pradesh                 47
Bihar                          46
Manipur                        44
Mizoram                        43
Andaman and Nicobar Islands    42
Goa                            42
Assam                          36
Jharkhand                      35
Arunachal Prad

In [391]:
covid_india.duplicated().sum()

0

In [392]:
covid_india.isna().sum()

Sno                         0
Date                        0
Time                        0
State/UnionTerritory        0
ConfirmedIndianNational     0
ConfirmedForeignNational    0
Cured                       0
Deaths                      0
Confirmed                   0
dtype: int64

In [393]:
covid_india.Deaths.max()

617

In [394]:
covid_india['State/UnionTerritory'].unique()

array(['Kerala', 'Telengana', 'Delhi', 'Rajasthan', 'Uttar Pradesh',
       'Haryana', 'Ladakh', 'Tamil Nadu', 'Karnataka', 'Maharashtra',
       'Punjab', 'Jammu and Kashmir', 'Andhra Pradesh', 'Uttarakhand',
       'Odisha', 'Puducherry', 'West Bengal', 'Chhattisgarh',
       'Chandigarh', 'Gujarat', 'Himachal Pradesh', 'Madhya Pradesh',
       'Bihar', 'Manipur', 'Mizoram', 'Andaman and Nicobar Islands',
       'Goa', 'Unassigned', 'Assam', 'Jharkhand', 'Arunachal Pradesh',
       'Tripura', 'Nagaland', 'Meghalaya', 'Nagaland#', 'Jharkhand#',
       'Dadar Nagar Haveli'], dtype=object)

In [395]:
covid_india_state_testing.sum()

Date            2020-04-172020-04-242020-04-272020-05-012020-0...
State           Andaman and Nicobar IslandsAndaman and Nicobar...
TotalSamples                                          1.57851e+07
Negative                                              1.30916e+07
Positive                                                   571595
dtype: object

In [396]:
covid_india_state_testing.duplicated().sum()

2

In [397]:
covid_india_state_testing.isna().sum()

Date              0
State             0
TotalSamples      0
Negative        136
Positive          9
dtype: int64

In [398]:
covid_india_state_testing.describe()

TotalSamples       Negative      Positive
count     759.000000     623.000000    750.000000
mean    20797.184453   21013.754414    762.126667
std     29959.945068   30047.832110   1542.251749
min        58.000000       0.000000      0.000000
25%      2392.500000    2492.500000     33.000000
50%      8612.000000    8310.000000    207.000000
75%     24910.000000   24462.500000    737.500000
max    175323.000000  162349.000000  14541.000000

********Assessing time_series Data of covid_19 All Countries Data .

In [399]:
covid_time_series_C.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264 entries, 0 to 263
Columns: 103 entries, Province/State to 4/29/20
dtypes: float64(2), int64(99), object(2)
memory usage: 212.6+ KB


In [400]:
covid_time_series_covid_19_R.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250 entries, 0 to 249
Columns: 103 entries, Province/State to 4/29/20
dtypes: float64(2), int64(99), object(2)
memory usage: 201.3+ KB


In [401]:
covid_time_series_D.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264 entries, 0 to 263
Columns: 103 entries, Province/State to 4/29/20
dtypes: float64(2), int64(99), object(2)
memory usage: 212.6+ KB


In [402]:
covid_time_series_C.isna().sum()

Province/State    182
Country/Region      0
Lat                 0
Long                0
1/22/20             0
                 ... 
4/25/20             0
4/26/20             0
4/27/20             0
4/28/20             0
4/29/20             0
Length: 103, dtype: int64

In [403]:
covid_time_series_covid_19_R.isna().sum()

Province/State    183
Country/Region      0
Lat                 0
Long                0
1/22/20             0
                 ... 
4/25/20             0
4/26/20             0
4/27/20             0
4/28/20             0
4/29/20             0
Length: 103, dtype: int64

In [404]:
covid_time_series_D.isna().sum()

Province/State    182
Country/Region      0
Lat                 0
Long                0
1/22/20             0
                 ... 
4/25/20             0
4/26/20             0
4/27/20             0
4/28/20             0
4/29/20             0
Length: 103, dtype: int64

In [405]:
covid_time_series_C.describe()

Lat        Long     1/22/20     1/23/20     1/24/20     1/25/20  \
count  264.000000  264.000000  264.000000  264.000000  264.000000  264.000000   
mean    21.317326   22.168315    2.102273    2.477273    3.564394    5.431818   
std     24.734994   70.669996   27.382118   27.480921   34.210982   47.612615   
min    -51.796300 -135.000000    0.000000    0.000000    0.000000    0.000000   
25%      6.969250  -20.026050    0.000000    0.000000    0.000000    0.000000   
50%     23.488100   20.535638    0.000000    0.000000    0.000000    0.000000   
75%     41.166075   78.750000    0.000000    0.000000    0.000000    0.000000   
max     71.706900  178.065000  444.000000  444.000000  549.000000  761.000000   

           1/26/20      1/27/20      1/28/20      1/29/20  ...        4/20/20  \
count   264.000000   264.000000   264.000000   264.000000  ...     264.000000   
mean      8.022727    11.087121    21.128788    23.356061  ...    9362.723485   
std      66.537101    89.647535   220.011922   221.352587  ...   53660.506404   
min       0.000000     0.000000     0.000000     0.000000  ...       0.000000   
25%       0.000000     0.000000     0.000000     0.000000  ...      59.250000   
50%       0.000000     0.000000     0.000000     0.000000  ...     343.500000   
75%       0.000000     0.000000     0.000000     0.000000  ...    1678.500000   
max    1058.000000  1423.000000  3554.000000  3554.000000  ...  784326.000000   

             4/21/20        4/22/20        4/23/20        4/24/20  \
count     264.000000     264.000000     264.000000     264.000000   
mean     9654.625000    9939.799242   10256.598485   10650.011364   
std     55395.197812   57150.737463   58974.532580   61245.507883   
min         0.000000       0.000000       0.000000       0.000000   
25%        64.000000      64.500000      71.500000      73.000000   
50%       352.500000     357.000000     380.500000     416.000000   
75%      1737.500000    1787.500000    1797.750000    1816.250000   
max    811865.000000  840351.000000  869170.000000  905358.000000   

             4/25/20        4/26/20        4/27/20       4/28/20       4/29/20  
count     264.000000     264.000000     264.000000  2.640000e+02  2.640000e+02  
mean    10975.848485   11258.950758   11521.833333  1.180454e+04  1.209805e+04  
std     63283.272344   65000.703417   66444.778601  6.800718e+04  6.972285e+04  
min         0.000000       0.000000       0.000000  0.000000e+00  0.000000e+00  
25%        73.000000      74.750000      75.750000  7.575000e+01  7.750000e+01  
50%       416.500000     432.500000     433.500000  4.435000e+02  4.615000e+02  
75%      1915.750000    1960.500000    2008.250000  2.048500e+03  2.068250e+03  
max    938154.000000  965785.000000  988197.000000  1.012582e+06  1.039909e+06  

[8 rows x 101 columns]

In [406]:
covid_time_series_covid_19_R.describe()

Lat        Long     1/22/20     1/23/20   1/24/20    1/25/20  \
count  250.000000  250.000000  250.000000  250.000000  250.0000  250.00000   
mean    20.048575   27.934869    0.112000    0.120000    0.1440    0.15600   
std     24.394560   67.432156    1.770875    1.774881    1.9664    2.03278   
min    -51.796300 -106.346800    0.000000    0.000000    0.0000    0.00000   
25%      6.677575   -8.091400    0.000000    0.000000    0.0000    0.00000   
50%     21.805100   22.380900    0.000000    0.000000    0.0000    0.00000   
75%     39.376275   87.379325    0.000000    0.000000    0.0000    0.00000   
max     71.706900  178.065000   28.000000   28.000000   31.0000   32.00000   

         1/26/20    1/27/20     1/28/20     1/29/20  ...       4/20/20  \
count  250.00000  250.00000  250.000000  250.000000  ...    250.000000   
mean     0.20800    0.24400    0.428000    0.504000  ...   2583.620000   
std      2.66555    2.86939    5.090034    5.599231  ...  11229.917142   
min      0.00000    0.00000    0.000000    0.000000  ...      0.000000   
25%      0.00000    0.00000    0.000000    0.000000  ...     12.000000   
50%      0.00000    0.00000    0.000000    0.000000  ...    121.000000   
75%      0.00000    0.00000    0.000000    0.000000  ...    577.500000   
max     42.00000   45.00000   80.000000   88.000000  ...  91500.000000   

            4/21/20       4/22/20        4/23/20        4/24/20  \
count    250.000000    250.000000     250.000000     250.000000   
mean    2721.560000   2842.008000    2957.636000    3160.324000   
std    11601.075828  12028.782034   12450.593102   13383.514668   
min        0.000000      0.000000       0.000000       0.000000   
25%       12.250000     15.250000      15.250000      17.250000   
50%      135.000000    144.000000     144.000000     155.500000   
75%      636.500000    669.750000     691.250000     762.750000   
max    95200.000000  99400.000000  103300.000000  109800.000000   

             4/25/20        4/26/20        4/27/20        4/28/20  \
count     250.000000     250.000000     250.000000     250.000000   
mean     3271.352000    3462.932000    3575.868000    3714.632000   
std     13643.759318   14647.648784   15055.674939   15514.122251   
min         0.000000       0.000000       0.000000       0.000000   
25%        18.000000      18.250000      19.000000      19.500000   
50%       158.000000     159.500000     170.000000     182.000000   
75%       803.000000     868.000000     925.750000     975.000000   
max    109800.000000  117727.000000  120832.000000  123903.000000   

             4/29/20  
count     250.000000  
mean     3890.876000  
std     16181.043925  
min         0.000000  
25%        21.500000  
50%       182.000000  
75%      1000.000000  
max    132929.000000  

[8 rows x 101 columns]

In [407]:
covid_time_series_D.describe()

Lat        Long     1/22/20     1/23/20     1/24/20     1/25/20  \
count  264.000000  264.000000  264.000000  264.000000  264.000000  264.000000   
mean    21.317326   22.168315    0.064394    0.068182    0.098485    0.159091   
std     24.734994   70.669996    1.046278    1.047853    1.479183    2.462894   
min    -51.796300 -135.000000    0.000000    0.000000    0.000000    0.000000   
25%      6.969250  -20.026050    0.000000    0.000000    0.000000    0.000000   
50%     23.488100   20.535638    0.000000    0.000000    0.000000    0.000000   
75%     41.166075   78.750000    0.000000    0.000000    0.000000    0.000000   
max     71.706900  178.065000   17.000000   17.000000   24.000000   40.000000   

          1/26/20     1/27/20     1/28/20     1/29/20  ...       4/20/20  \
count  264.000000  264.000000  264.000000  264.000000  ...    264.000000   
mean     0.212121    0.310606    0.496212    0.503788  ...    655.757576   
std      3.201783    4.678456    7.693263    7.693757  ...   3718.717575   
min      0.000000    0.000000    0.000000    0.000000  ...      0.000000   
25%      0.000000    0.000000    0.000000    0.000000  ...      1.000000   
50%      0.000000    0.000000    0.000000    0.000000  ...      6.000000   
75%      0.000000    0.000000    0.000000    0.000000  ...     42.250000   
max     52.000000   76.000000  125.000000  125.000000  ...  42659.000000   

            4/21/20       4/22/20       4/23/20       4/24/20       4/25/20  \
count    264.000000    264.000000    264.000000    264.000000    264.000000   
mean     682.791667    708.068182    733.625000    757.590909    781.000000   
std     3881.003664   4032.269842   4182.178953   4308.310233   4453.413482   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        1.000000      1.000000      1.000000      1.000000      1.000000   
50%        6.500000      7.000000      7.000000      7.000000      7.000000   
75%       43.500000     44.500000     45.500000     49.500000     51.500000   
max    45086.000000  47412.000000  49724.000000  51493.000000  53755.000000   

            4/26/20       4/27/20       4/28/20       4/29/20  
count    264.000000    264.000000    264.000000    264.000000  
mean     795.041667    812.378788    836.276515    862.265152  
std     4530.812519   4624.558159   4762.356681   4925.301761  
min        0.000000      0.000000      0.000000      0.000000  
25%        1.000000      1.000000      1.000000      1.000000  
50%        7.000000      7.000000      7.500000      7.500000  
75%       54.250000     57.250000     58.000000     60.250000  
max    54881.000000  56259.000000  58355.000000  60967.000000  

[8 rows x 101 columns]

#### Data Issue 5:
***Drop [Sno.] from column in Covid_india_age dataframe.

In [408]:
covid_india_age = covid_india_age.drop(['Sno'], axis=1)

In [409]:
covid_india_age

AgeGroup  TotalCases Percentage
0      0-9          22      3.18%
1    10-19          27      3.90%
2    20-29         172     24.86%
3    30-39         146     21.10%
4    40-49         112     16.18%
5    50-59          77     11.13%
6    60-69          89     12.86%
7    70-79          28      4.05%
8     >=80          10      1.45%
9  Missing           9      1.30%

#*** Covid_india['State/UnionTerritory']csv file assign the value in nagaland two time .these issue is negelected for analysis .

#### Data Issue 6:
*** Missing value (NaN)in 3 covid_time_series dataframe:Confirmed,Recovered,Death. 

In [410]:
covid_time_series_covid_19_R.fillna(0, inplace=True)

In [411]:
covid_time_series_C.fillna(0,inplace = True)

In [412]:
covid_time_series_D.fillna(0,inplace =True)

#### Data Issue 7:
***Drop [province/stste, lat,long] from column in Covid time_series dataframe.

In [ ]:
covid_time_series_covid_19_R = covid_time_series_covid_19_R.drop(['Province/State','Lat','Long'], axis=1)
covid_time_series_covid_19_R 


In [ ]:
covid_time_series_C =covid_time_series_C.drop(['Province/State','Lat','Long'], axis=1)
#covid_time_series_C

In [ ]:
covid_time_series_D = covid_time_series_D.drop(['Province/State','Lat','Long'], axis=1)
#covid_time_series_D

#### Data Issue 8:
***  Difficult to analyze time_series data. In this dataframe analysis melt funtion is  used .Unpivot date columns[2:] with variable column ‘Date’ and value column ‘Confirmed’,'Recovered','Death'

In [ ]:
dates = covid_time_series_C.columns[2:]
covid_time_series_C = covid_time_series_C.melt(
    id_vars=['Country/Region'], 
    value_vars=dates, 
    var_name='Date', 
    value_name='Confirmed'

)

covid_time_series_C


In [ ]:
dates = covid_time_series_D.columns[2:]
covid_time_series_D = covid_time_series_D.melt(
    id_vars=['Country/Region'], 
    value_vars=dates, 
    var_name='Date', 
    value_name='Deaths'
)

covid_time_series_D

In [ ]:

dates = covid_time_series_covid_19_R.columns[2:]
covid_time_series_covid_19_R= covid_time_series_covid_19_R.melt(
    id_vars=['Country/Region'], 
    value_vars=dates, 
    var_name='Date', 
    value_name='Recovered'
    
)

covid_time_series_covid_19_R

#### Data Issue 9:
*Merge the Covid_time_series_C,Covid_time_series_D,Covid_time_series_covid_19_R using merge function.
*Calculate Death_percentage and Recovered_percentage in Covid_time_series dataframe.

In [ ]:
covid_time_series= covid_time_series_C.merge(right=covid_time_series_D, how='left',on=['Country/Region', 'Date'])
covid_time_series = covid_time_series.merge( right=covid_time_series_covid_19_R, how='left',on=['Country/Region', 'Date'])
covid_time_series['Death_percentage'] = (covid_time_series.Deaths / covid_time_series.Confirmed)/100
covid_time_series['Recovered_percentage'] = (covid_time_series.Recovered / covid_time_series.Confirmed)/100
covid_time_series.duplicated().sum()
covid_time_series = covid_time_series.drop_duplicates() 

In [ ]:
covid_time_seriesChina = covid_time_series[covid_time_series['Country/Region'] == 'China']
covid_time_seriesChina.head(50)

In [ ]:
def getfirst_iterrows_loop(df):
    for index, row in df.iterrows():
        if (row['Confirmed'] == 1):
            return row['Date']
    return None
    
    
df3=covid_time_series.groupby(['Country/Region'])['Country/Region','Confirmed','Date'].apply(getfirst_iterrows_loop).reset_index()
df3.info()


In [ ]:
df3.rename(columns={ df3.columns[1]: "firstConfDate" }, inplace = True)


In [ ]:
from datetime import datetime
def setFirstConfDate(record):
    days = 0
    for index, row in df3.iterrows():
        if (row['Country/Region'] == record['Country/Region'] and row['firstConfDate'] != None):
            delta = datetime.strptime(record['Date'], '%m/%d/%y').date() - datetime.strptime(row['firstConfDate'], '%m/%d/%y').date()
            days = delta.days
    print(record['Country/Region'] + ' ' + record['Date'] + ' ' + row['firstConfDate'] + ' ' + str(days))
    return days
    
covid_time_series['firstConfDate'] =covid_time_series.apply(lambda row: setFirstConfDate(row), axis=1)


In [ ]:
covid_time_series

#### Data Issue10:
*** Replace NaN value with Zero in Death_Percentage ,Recovered_percentage in covid_time_series.

In [ ]:
covid_time_series.fillna(0, inplace=True)

In [ ]:
covid_time_series['Date'] = pd.to_datetime(covid_time_series['Date'], errors='coerce')
covid_time_series['Day'] = covid_time_series['Date'].dt.day
covid_time_series['Day']

In [ ]:
date  =  dt.covid_time_series(202)
month_first_day = date.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
print(month_first_day)

*** Store Covid_time_series data in New csv file:

In [ ]:
covid_time_series.to_csv('covid_time_series1.csv')

In [ ]:
covid_time_series_I = covid_time_series[covid_time_series['Country/Region']=='India']

covid_time_series_I.head(5)

### Exploratory data analysis (EDA)
 
######### Covid_19 All countries Data:
Determine mortality rates and recovered cases of covid 19,Analysis is based on the number of excess deaths in different countries and predicted the mortality and Recovered patients in different country.

***This plot shows countrywise death cases on a logarithmic scale.

In [ ]:
fig = px.scatter(full_data,x="total_cases",y="total_deaths",text="country",log_x=True ,log_y=True ,size_max=100, range_x=[1,1000000000],range_y=[1,1000000])
fig.update_traces(textposition='top center')
fig.update_layout(
    height=800,width=1000,
    title_text='Total Deaths Cases in the world',xaxis = dict(
        tickangle = 90,
        title_text = "Total_cases",
        title_font = {"size": 15},
        title_standoff = 10),
    yaxis = dict(
        title_text = "Total_Deaths Cases",
        title_standoff = 10)
)
fig.show()

- Total Deaths Reported in many countries below 1000 

***This plot shows countrywise recoverd cases on a logarithmic scale.
**** here all countries is consider for analysis so, there is overlapping in scatter plot.

In [ ]:
fig = px.scatter(full_data,x="total_cases",y="total_recovered", text="country",  log_x=True ,log_y=True ,size_max=100, range_x=[1,10000000],range_y=[1,1000000])
fig.update_traces(textposition='top center')
fig.update_layout(
    height=800,width=1000,
    title_text='Total Recovered Cases in the world',
    xaxis = dict(
        tickangle = 90,
        title_text = "Total_cases",
        title_font = {"size": 15},
        title_standoff = 10),
    yaxis = dict(
        title_text = "Total_recovered Cases",
        title_standoff = 10)

    
    
)
fig.show()

### Analysis of Covid_19 Cases in All countries Based on Statsmodel:
-Statsmodels module is used for covid_19 cases that provides classes and functions for the estimation  regression models,for conducting statistical tests, and statistical data exploration of covid_19('total_cases','total_deaths','new_cases','new_deaths','total_recovered','active_cases')in all countries.  
-Assumptions of a regression model:
1. Relevance of data to the covid_19 cases. 
2. Linearity and additivity 
3. Independence of residuals 
4. Constancy of variance of residuals 
5. Normal distribution of residuals

In [ ]:
X = full_data[['total_deaths','new_cases','new_deaths']] 
# #### fit a OLS model with intercept on total_cases and  new_cases,new_deaths.
y = full_data['total_cases']
X = sm.add_constant(X)
est = sm.OLS(y, X).fit()
print(est.summary())


*β₀ and β₁ are chosen to minimize the square of the distance between the predicted values and the actual values.

###Adj. R-squared indicates that 99% of total_deaths,new_cases,new_deaths can be explained by our predictor variable.

In order to understand trends we look at the slope of the death cases, new cases and total deaths in linearscale..

From our results, we see that
• The intercept 𝛽̂0 = -111.27

The regression coefficient (coef) represents the change in the dependent variable resulting from a one unit change in the predictor variable, all other variables being held constant. In our model, a one unit increase in total_deaths, new_cases,new_deaths increase total_cases.

• The slope 𝛽̂1 =  5.0908

• The slope 𝛽̂2 = 11.7764 .

• The slope 𝛽̂3 = 321.7351

• The positive 𝛽̂3 parameter estimate implies high new_deaths rate 
In line with our assumptions, an increase in total_deaths, new_cases, new_deaths  appears to increase the total cases.


The p-value means the probability of an 5.0908 increase in total_cases  due to a one unit increase in total_deaths is 0%, assuming there is no relationship between the two variables.
##• The p-value of total_deaths and new_cases,new_deaths 0.000 for 𝛽̂1implies that is statistically significant (using p < 0.05 as a rejection rule).
##• The p-value of new_deaths is 0.03

**** The standard error measures the accuracy of total_deaths coefficient by estimating the variation of the coefficient if the same test were run on a different sample . Our standard error 1, is low and therefore appears accurate.

##• The R-squared value of R-squared:0.994.

In [ ]:
fig = plt.figure(figsize =(15,8))
results = smf.ols('total_cases~total_deaths+new_cases+new_deaths',data = full_data).fit()
sm.graphics.plot_regress_exog(results, 'total_deaths', fig=fig)
plt.show()


1. The “Y and Fitted vs. X” graph plots the dependent variable against our predicted values with a confidence interval. The inverse relationship in our graph indicates that total_cases  and total_deaths are positively correlated, i.e., when one variable increases the other increase.

2.The “Residuals versus total_deaths graph shows our model's errors versus the specified predictor variable. Each dot is an observed value; the line represents the mean of those observed values.

3.The “Partial regression plot” shows the relationship between total_cases and total_deaths,the impact of adding other independent variables on our existing total_deaths coefficient.

4.the Component and Component Plus Residual (CCPR) plot is an extension of the partial regression plot, but shows where our trend line would lie after adding the impact of adding our other independent variables on our existing total_deaths coefficient.This is the "component" part of the plot and is intended to show where the "fitted line" would lie.

In [ ]:
X = full_data[['total_recovered','new_cases','active_cases']] 
#### fit a OLS model with intercept on total_recovered and  new_cases,active_cases.
y = full_data['total_cases']
X = sm.add_constant(X)
est = sm.OLS(y, X).fit()
print(est.summary())


 *β₀ and β₁ are chosen to minimize the square of the distance between the predicted values and the actual values.

###Adj. R-squared indicates that 100% of total_recovered,new_cases,active_cases can be explained by our predictor variable.

In order to understand trends we look at the slope of the recoverd_cases, new cases and active_case in linearscale..

From our results, we see that
• The intercept 𝛽̂0 = 3.0684
The regression coefficient (coef) represents the change in the dependent variable resulting from a one unit change in the predictor variable, all other variables being held constant. In our model, a one unit increase the recoverd_cases, new cases and active_case increase total_cases.

• The slope 𝛽̂1 = 1.0937.

• The slope 𝛽̂2 = 0.1470

• The slope 𝛽̂3 = 1.0588.

• The positive 𝛽̂2 parameter estimate implies low new_cases rate 
In line with our assumptions, an increase in the recoverd_cases,active_case  appears to increase the total cases.


The p-value means the probability of an 1.1121 increase in total_cases  due to a one unit increase in total_recovered is 0%, assuming there is no relationship between the two variables.
##• The p-value of total_deaths and new_cases,new_deaths 0.000 for 𝛽̂1implies that is statistically significant (using p < 0.05 as a rejection rule).

**** The standard error measures the accuracy of total_deaths coefficient by estimating the variation of the coefficient if the same test were run on a different sample . Our standard error, 0.007, is low and therefore appears accurate.

##• The R-squared value of R-squared:1.

In [ ]:
fig = plt.figure(figsize =(15,8))
results = smf.ols('total_cases~total_recovered  + new_cases+active_cases', data = full_data).fit()
#sm.graphics.plot_ccpr_grid(results, fig=fig)
sm.graphics.plot_regress_exog(results, 'total_recovered', fig=fig)
plt.show()
###endogenous: caused by factors within the system ,exogenous: caused by factors outside the system

1. The “Y and Fitted vs. X” graph plots the dependent variable against our predicted values with a confidence interval. The inverse relationship in our graph indicates that total_cases  and total_recovered cases  are positively correlated, i.e., when one variable increases the other increase.

2.The “Residuals versus total_deaths graph shows our model's errors versus the specified predictor variable. Each dot is an observed value; the line represents the mean of those observed values.

3.The “Partial regression plot” shows the relationship between total_cases and total_recovered,the impact of adding other independent variables on our existing total_recovered coefficient.

4.the Component and Component Plus Residual (CCPR) plot is an extension of the partial regression plot, but shows where our trend line would lie after adding the impact of adding our other independent variables on our existing total_recovered coefficient.This is the "component" part of the plot and is intended to show where the "fitted line" would lie.


### Analysis of Covid_19  Total_cases ,Active_Case, Serious Cases  in All countries.

In [ ]:
fig = px.scatter(full_data, y='total_cases', x='active_cases',animation_frame="active_cases",text = "country",range_x =[0,100000],range_y=[0,100000])
fig2 = px.scatter(full_data, y='total_cases', x='serious', animation_frame="serious",text ='country',range_x =[0,10000],range_y=[0,100000]) 
fig.show()
fig2.show()

### Analysis of Covid_19  Totcase/1Mpop,Deaths/1Mpop,Tests/1Mpop in All countries.


In [ ]:
fig = px.scatter(full_data, x='tot\xa0cases/1m_pop', y='deaths/1m_pop', color='country',log_x=True ,log_y=True ,size_max=45)
fig2 = px.scatter(full_data, x='tot\xa0cases/1m_pop', y='tests/_1m_pop', color='country',log_x=True ,log_y=True ,size_max=45) 
fig.show()
fig2.show()


####   World  Maps of Total_covid 19 Cases in Countries:
  *  Total cases Maps using  Plotly ,Mapbox and animation Show how the covid19 Spread in countries.
 

In [ ]:
fig = px.scatter_mapbox(full_data, lat="lat", lon="long",color = 'country', hover_name="country", hover_data=['total_cases', "total_deaths"],
                        color_continuous_scale=px.colors.cyclical.IceFire,
                        animation_frame='total_cases',size_max=55, zoom=3, height=500)
fig.update_layout(title_text="Total_Cases in World")
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

####   World  Maps of Total_Deaths Cases in All Countries:
  *  Total deaths Maps using  Plotly ,Mapbox .

In [ ]:
fig = px.scatter_mapbox(full_data, lat="lat", lon="long",color = 'country', hover_name="country", hover_data=['total_cases',"total_deaths" ,"total_recovered"],
                        color_continuous_scale=px.colors.cyclical.IceFire,
                        animation_frame='total_deaths',zoom=3, height=500)
fig.update_layout(title_text="Total_Deaths in World")
fig.update_layout(
 mapbox_style="white-bg",
    mapbox_layers=[
        {
            "below": 'traces',
            "sourcetype": "raster",
            "source": [
                "https://basemap.nationalmap.gov/arcgis/rest/services/USGSImageryOnly/MapServer/tile/{z}/{y}/{x}"
            ]
        }
      ])
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

In [ ]:
full_data = full_data.sort_values(by=['tests/_1m_pop'])
full_data

In [ ]:

full_data=pd.melt(full_data, id_vars=['country','tests/_1m_pop'], value_vars=['total_cases', 'new_cases', 'total_deaths', 'new_deaths', 'total_recovered'])
# plotly 
fig = px.line(full_data, x='country', y='value', color='variable',log_x=False ,log_y=True)

# Show plot 
fig.show()

##### Exploratory data analysis (EDA) INDIA:
######### Covid_19 Cases in India:
Determine mortality rates and Cured cases of covid 19,Analysis is based on the number of excess deaths in India and predicted the mortality and Recovered patients in State and Union Territory .


### Univariate analysis:

In [ ]:
base_color = sns.color_palette()[1]
plt.figure(figsize=(32,6))
g = sns.countplot(data = covid_india, x ='State/UnionTerritory', color = base_color)
g.set_xticklabels(g.get_xticklabels(), rotation=45)
n_points = covid_india.shape[0]
cat_counts = covid_india['State/UnionTerritory'].value_counts()
locs, labels = plt.xticks() 
for loc, label in zip(locs, labels):
    count = cat_counts[label.get_text()]
    pct_string = '{:0.1f}%'.format(100*count/n_points)
    plt.text(loc, count-8, pct_string, ha = 'left',va='bottom', color = 'black')

### BiVariate Analysis of Covid_19 Cases in  India:

In [ ]:
fig = px.scatter(covid_india,x="Confirmed",y="Deaths" ,animation_frame="Deaths", animation_group="State/UnionTerritory",color="State/UnionTerritory",log_x=True ,log_y=True , range_x=[1,10000],range_y=[1,10000])
fig.update_traces(textposition='top center')

fig.update_layout(
    height=800,width=1000,
    title_text='Total Deaths Cases in India',xaxis = dict(
        tickangle = 90,
        title_text = "Total_cases",
        title_font = {"size": 15},
        title_standoff = 10),
    yaxis = dict(
        title_text = "Total_Deaths Cases",
        title_standoff = 10),
    
)

fig.show()

In [ ]:

fig = px.scatter(covid_india,x="Confirmed",y="Cured", animation_frame="Cured", animation_group="State/UnionTerritory",color="State/UnionTerritory", log_x=True ,log_y=True ,size_max=45, range_x=[1,10000],range_y=[1,10000])
fig.update_traces(textposition='top center')

fig.update_layout(
    height=800,width=1000,
    title_text='Total Recovered Cases in India',xaxis = dict(
        tickangle = 90,
        title_text = "Total_cases",
        title_font = {"size": 15},
        title_standoff = 10),
    yaxis = dict(
        title_text = "Total Recovered Cases",
        title_standoff = 10)
)


fig.show()

### Analysis of Covid_19 Cases in  India Based on StatsModel:
-Statsmodels module is used for covid_19 cases that provides classes and functions for the estimation  regression models,for conducting statistical tests, and statistical data exploration of covid_19('Confirmed cases','Deaths cases','Cured cases')in india. 

In [ ]:
X = covid_india[['Deaths']] 
#### fit a OLS model with intercept on Deaths
y = covid_india['Confirmed']
X = sm.add_constant(X)
est = sm.OLS(y, X).fit()
print(est.summary())


*β₀ and β₁ are chosen to minimize the square of the distance between the predicted values and the actual values.

###Adj. R-squared indicates that 89% of deaths can be explained by our predictor variable.

In order to understand trends we look at the slope of the death cases in linearscale..

From our results, we see that
• The intercept 𝛽̂0 = 113
The regression coefficient (coef) represents the change in the dependent variable resulting from a one unit change in the predictor variable, all other variables being held constant. In our model, a one unit increase in deaths, new_cases,increase Confirmed _cases.
• The slope 𝛽̂1 = 22.66
In line with our assumptions, an increase in deaths appears to increase the confirmed cases.


The p-value means the probability of an 22.66 increase in Confirmed_cases  due to a one unit increase in deaths is 0%, assuming there is no relationship between the two variables.
##• The p-value of deaths is  0.000 for 𝛽̂1implies that is statistically significant (using p < 0.05 as a rejection rule).

**** The standard error measures the accuracy of deaths coefficient by estimating the variation of the coefficient if the same test were run on a different sample . Our standard error, 0.195, is low and therefore appears accurate.

##• The R-squared value of R-squared:0.89 .

In [ ]:

fig = plt.figure(figsize =(15,8))
#full_data1= sm.dataset.full_data.load_pandas()
results = smf.ols('Confirmed ~Deaths ', data = covid_india).fit()
#sm.graphics.plot_ccpr_grid(results, fig=fig)
sm.graphics.plot_regress_exog(results, 'Deaths', fig=fig)
plt.show()

1. The “Y and Fitted vs. X” graph plots the dependent variable against our predicted values with a confidence interval. The inverse relationship in our graph indicates that Confirmed_cases  and  deaths are positively correlated, i.e., when one variable increases the other increase.

2.The “Residuals versus total_deaths graph shows our model's errors versus the specified predictor variable. Each dot is an observed value; the line represents the mean of those observed values.

3.The “Partial regression plot” shows the relationship between Confirmed_cases  and  deaths,the impact of adding other independent variables on our existing deaths coefficient.

4.the Component and Component Plus Residual (CCPR) plot is an extension of the partial regression plot, but shows where our trend line would lie after adding the impact of adding our other independent variables on our existing deaths coefficient.This is the "component" part of the plot and is intended to show where the "fitted line" would lie.

In [ ]:
X = covid_india[['Cured']] 
#### fit a OLS model with intercept on Cured Cases
y = covid_india['Confirmed']
X = sm.add_constant(X)
est = sm.OLS(y, X).fit()
print(est.summary())


*β₀ and β₁ are chosen to minimize the square of the distance between the predicted values and the actual values.

###Adj. R-squared indicates that 79% of Cured Cases can be explained by our predictor variable.

In order to understand trends we look at the slope of the Cured Cases in linearscale..

From our results, we see that
• The intercept 𝛽̂0 = 42.61
The regression coefficient (coef) represents the change in the dependent variable resulting from a one unit change in the predictor variable, all other variables being held constant. In our model, a one unit increase in Cured increase Confirmed.
• The slope 𝛽̂1 = 4.21
In line with our assumptions, an increase in cured cases  appears to increase the  Confirmed cases.


The p-value means the probability of an 4.21 increase in Confirmed cases  due to a one unit increase in Cured Cases is 0%, assuming there is no relationship between the two variables.
##• The p-value of cured  0.000 for 𝛽̂1implies that is statistically significant (using p < 0.05 as a rejection rule).


**** The standard error measures the accuracy of total_deaths coefficient by estimating the variation of the coefficient if the same test were run on a different sample . Our standard error, 0.050, is low and therefore appears accurate.

##• The R-squared value of R-squared:0.79 .

In [ ]:
fig = plt.figure(figsize =(15,8))
#full_data1= sm.dataset.full_data.load_pandas()
results = smf.ols('Confirmed ~Cured', data = covid_india).fit()
#sm.graphics.plot_ccpr_grid(results, fig=fig)
sm.graphics.plot_regress_exog(results, 'Cured', fig=fig)
plt.show()

1. The “Y and Fitted vs. X” graph plots the dependent variable against our predicted values with a confidence interval. The inverse relationship in our graph indicates that Cured and Confirmed are positively correlated, i.e., when one variable increases the other increase.

2.The “Residuals versus cured  graph shows our model's errors versus the specified predictor variable. Each dot is an observed value; the line represents the mean of those observed values.

3.The “Partial regression plot” shows the relationship between Cured and confirmed cases ,the impact of adding other independent variables on our existing  cured coefficient.

4.the Component and Component Plus Residual (CCPR) plot is an extension of the partial regression plot, but shows where our trend line would lie after adding the impact of adding our other independent variables on our existing cured coefficient.This is the "component" part of the plot and is intended to show where the "fitted line" would lie.

### Analysis of Covid_19 Cases in  India and its State /Union Territory :

In [ ]:
f,g = plt.subplots(figsize = (15,10))
base_color = sns.color_palette()[0]
g = sns.barplot(data = covid_india, x = 'Confirmed', y = 'State/UnionTerritory',
            label = 'Total Confirmed Cases',color = base_color)
sns.set_color_codes('muted')
g = sns.barplot(x = 'Cured', y = 'State/UnionTerritory', data = covid_india,
            label = 'Total number of Cured', color = 'R', edgecolor = 'w')
sns.set_color_codes('pastel')
g= sns.barplot(x = 'Deaths', y = 'State/UnionTerritory', data = covid_india,
            label = 'Total number of Deaths', color = 'g', edgecolor = 'w')
g.legend(ncol = 3, loc = 'lower right')
#sns.despine(left = True, bottom = True)
plt.show()

### Analysis of Covid_19 Cases in  India and its State /Union Territory

In [ ]:
plt.figure(2, figsize=(20,15))
fig,ax = plt.subplots(1, 2)
g =sns.scatterplot(data=covid_india,y="Deaths",x="Confirmed",ax=ax[0],hue="State/UnionTerritory",palette="deep")
g.legend(loc='upper right', bbox_to_anchor=(.20, 0.0), ncol=1)
g =sns.scatterplot(data=covid_india,y="Cured",x="Confirmed",ax=ax[1],hue="State/UnionTerritory",palette="deep")
g.legend(loc='upper left', bbox_to_anchor=(1.0, 0.0), ncol=1)

### Analysis Based on Age group in india

In [ ]:

fig = px.bar(covid_india_age, y='TotalCases', x='AgeGroup', text='Percentage')
fig.update_traces(texttemplate='%{text}', textposition='outside',marker_color='lightsalmon')
fig.update_layout(uniformtext_minsize=8, uniformtext_mode='hide')
fig.show()

In [ ]:
s = covid_india_state_testing.sum()
s

### Analysis of Covid_19 Cases Based  on  Testing (Positive Cases and Negative Case) in  India :

In [ ]:
fig = px.scatter(covid_india_state_testing, x='Date', y='Positive', title='Positive Cases Time Series with Rangeslider', range_x=['1/23/20','5/4/20'],range_y = [1,100000])
fig.update_traces(marker_color='indianred')
fig.update_xaxes(rangeslider_visible=True)
fig.show()
fig1 = px.scatter(covid_india_state_testing, x='Date', y='Negative', title='Negative Cases Time Series with Rangeslider',range_x = ['1/23/20','5/4/20'],range_y = [1,1000000])
fig1.update_traces(marker_color='lightsalmon')
fig1.update_xaxes(rangeslider_visible=True)
fig1.show()

## Covid_Time_Series Analysis: Datewise Confirmed Cases,Deaths, Recovered Cases in All Countries :
Date:22/jan/2020 till 29/April/2020
1. The total number of confirmed cases to date for each country and province.
2. The percentage of confirmed cases,Deaths,Recovered cases.
3. A maximum confirmed cases in all countries.
4. A maximum deaths cases in all countries.
5. A maximum recovered cases in all countries

In [ ]:
fig = px.scatter(covid_time_series_C, x='Date', y='Confirmed',color="Country/Region", title='Confirmed Cases Time Series with Rangeslider',range_x=['1/22/20','4/29/20'],range_y = [1,1000000])
fig.update_traces(marker_color='darkslateblue')
fig.update_xaxes(rangeslider_visible=True)
fig.show()

In [ ]:
fig = px.scatter(covid_time_series_D, x='Date', y='Deaths',color="Country/Region", title='Deaths Cases Time Series with Rangeslider',range_x=['1/22/20','4/29/20'],range_y = [1,100000])
fig.update_traces(marker_color='sandybrown')
fig.update_xaxes(rangeslider_visible=True)
fig.show()

In [ ]:
fig = px.scatter(covid_time_series_covid_19_R, x='Date', y='Recovered',color="Country/Region", title='Recovered Cases Time Series with Rangeslider',range_x=['1/22/20','4/29/20'],range_y = [1,1000000])
fig.update_traces(marker_color='green')
fig.update_xaxes(rangeslider_visible=True)
fig.show()

#### Daywise Covid_19 Analysis:
Countries had the highest number of confirmed cases , Deaths Cases and recovered cases in one particular day.

In [ ]:

fig = px.scatter(covid_time_series, x='Day', y='Confirmed',color="Country/Region",log_x=False ,log_y=True ) 
fig.show()
fig1 = px.scatter(covid_time_series, x='Day', y='Deaths',color="Country/Region",log_x=False ,log_y=True ) 
fig1.show()
fig2 = px.scatter(covid_time_series, x='Day', y='Recovered',color="Country/Region",log_x=False ,log_y=True )
fig2.show()

#### Conclusion: 
#######Analyze curated data related to the Covid_19 Viral Disease  to understand the magnitude of the risk posed by this infectious Disease in all Countries:
     - First cases of covid_19 is reported in Wuhan (the city where the virus originated)Central China, with a population of          over 11 million people.The city, on January 23.
     - 2-14 days represents the current official estimated range for the novel coronavirus COVID-19.
     - On January Month , the novel coronavirus cases in the UK,Russia,Sweden, Spain were reported less in number.
     - On March and April Months there is tremendous increase of Confirmed cases and Deaths Cases in all over World.
       ** Total Confirmed Case in World = 3795667
       ** Total Deaths Case in World = 
       ** Total Recovered Case in World=
       ** Total Confirmed Case in India = 
       ** Total Deaths Case in India = 
       ** Total Recovered Case in India =
       
- Age and conditions of Coronavirus cases in india  reported 
TotalSamples                                          1.57851e+07
Negative                                              1.30916e+07
Positive                                                   571595
   
    
- Time series data show that their increase in confirmed cases rapidly .and also the  death rate increase in countries .

- Time Series data also show that recovered cases  more than the death cases in countries . 

- Confirmed refers to a case being reported in contrast to a case being infected. Therefore, fluctuations in days .This is       due to the sampling bias induced by the limited amount of corona test kits.
    
 -So many cases that do not yet show symptoms are tested and the mortality rate below is shown in dead per million inhabitants.

    
     

In [ ]:
### Reference:
1.https://www.kaggle.com/sudalairajkumar/covid19-in-india
2.https://www.worldometers.info/coronavirus/
3.https://data.humdata.org/dataset/5dff64bc-a671-48da-aa87-2ca40d7abf02